tests: sched: Add busy threads for SMP

ycsin opened the pull request:
Basically the same idea as #77038

The sched benchmark is designed for systems with a single CPU. Otherwise, the timestamps would be wrong when the partner thread is scheduled on another CPU, i.e. negative values:

```
unpend   63 ready   62 switch -16562 pend 18937 tot 2500 (avg  928)
```

When the system allows for multiple CPUs, spawn a non-preemptible thread to keep the other CPUs busy.

ycsin committed:
tests: sched: Add busy threads for SMP

The sched benchmark is designed for systems with a single
CPU. Otherwise, the timestamps would be wrong when the partner
thread is scheduled on another CPU, i.e. negative values:

```
unpend   63 ready   62 switch -16562 pend 18937 tot 2500 (avg  928)
```

When the system allows for multiple CPUs, spawn a non-preemptible
thread to keep the other CPUs busy.

Signed-off-by: Yong Cong Sin <ycsin@meta.com>

peter-mitsis commented:
I just remembered about something. Adding the busy threads can really slow down some simulators (such as the intel_adsp_xxx simulators. Can you please add similar board overlays proposed for the latency_measure benchmark (see https://github.com/zephyrproject-rtos/zephyr/pull/80255/commits/ff3a8d9286c7fc3dd306ba01264633250ae0613f) for the sched benchmark?

ycsin committed:
tests: benchmarks/sched: limit CPU to 1 for Intel ADSP ACE

Due to addition of busy threads running on other cores, and
the simulator runs in single thread bouncing through all cores,
we are wasting quite a bit of time just busy waiting. This makes
each simulator run too long for CI. So limit CPU number to 1.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>
Signed-off-by: Yong Cong Sin <ycsin@meta.com>

ycsin commented:
stole https://github.com/zephyrproject-rtos/zephyr/commit/ff3a8d9286c7fc3dd306ba01264633250ae0613f, cc @dcpleung
